### PR TITLE
Add deterministic contribution IDs and stack lookup IDs for resolved artifacts

### LINKED ISSUE #4210: [Feature]: Add deterministic contribution IDs and stack lookup IDs for resolved artifacts
### Problem Statement

Spec Kit identifies contributed commands, templates, scripts, and hooks primarily by `name`. Names can collide across artifact kinds and source layers, and a name alone cannot reliably link a resolved artifact-stack layer to the exact manifest contribution that supplied it. Consumers such as `preset info --json`, `extension info --json`, and `specify artifact` need stable contribution identifiers and a join key that works across reinstalls and machines.

### Proposed Solution

Introduce computed, opaque `id` fields for every command, template, script, and hook returned by public preset/extension manifest and info APIs, plus a `lookupId` field on every provenance-backed non-built-in resolved artifact-stack layer.

For named preset, extension, and project-override contributions, derive IDs using:

```text
{layer}:{sourceId}:{kind}:{name}
```

Use `project`, `preset`, or `extension` for `layer`; `_` only for the project-override source ID; and the preset or extension manifest ID for manifest-declared sources. Use `command`, `template`, or `script` for `kind`. Examples include `preset:speckit.core:command:speckit.plan`, `extension:speckit.git:template:pr-body`, and `project:_:template:spec-template`.

For hooks, use `{eventName}:{command}` as the name component. Hooks are valid only for preset and extension layers. If duplicate event/command hook entries are valid, add an appropriate stable discriminator or reject duplicates; do not use array position.

Built-in artifacts do not have an originating manifest contribution and therefore do not receive a contribution `lookupId`. Every artifact, including built-ins, instead has a source-agnostic public ID of the form `{kind}:{name}`. Built-in stack rows are recognized by absent provenance fields (`layer`, `sourceId`, and `lookupId` are null), and round-trip through the public artifact ID.

Compute IDs at read, serialization, or resolution time rather than persisting them in authored or installed manifests. For manifest-declared preset and extension layers, `lookupId` must exactly match the originating contribution's `id`. Preserve existing name-based behavior and document IDs as opaque strings.

### Alternatives Considered

Install-time UUIDs are unsuitable because they differ across machines and reinstalls. Existing names are insufficient because they collide across sources and kinds. Content hashes are unsuitable because IDs would change whenever artifact content is edited. Array indexes are unsuitable for hooks because reordering entries would change their IDs.

A synthetic `core:_:...` contribution ID was considered for built-in assets. It was rejected because built-ins have no originating preset or extension manifest contribution to join to; the source-agnostic `{kind}:{name}` artifact ID provides their stable round-trip key without overloading `lookupId`.

Do not use the existing integration manifest as the ID source; it tracks installed file hashes and paths rather than manifest contribution identity.

### Component

Specify CLI (initialization, commands)

### AI Agent (if applicable)

_No response_

### Use Cases

1. A wizard can read an artifact stack and follow each provenance-backed layer's `lookupId` to the full contribution detail without re-parsing manifests.
2. Developers on different machines can use identical IDs when reporting or diagnosing a contribution.
3. Tooling can hash or cache resolved compositions using public artifact IDs plus the available layer `lookupId` values.
4. Future JSON output for `preset info`, `extension info`, and `specify artifact` can expose consistent cross-references.

### Acceptance Criteria

- [ ] Document the `layer:sourceId:kind:name` contribution grammar, the source-agnostic `kind:name` artifact ID, and the `lookupId` relationship in `extensions/EXTENSION-API-REFERENCE.md` and the preset API/manifest reference.
- [ ] Public preset/extension manifest and info representations expose computed `id` values for commands, templates, scripts, and hooks.
- [ ] Hook IDs are deterministic and collision-free without install paths or list indexes.
- [ ] Manifest-declared preset and extension artifact-stack layers expose `lookupId` equal to the corresponding contribution's `id`.
- [ ] Project-local override layers use the synthetic `project:_:{kind}:{name}` lookup form and intentionally have no manifest contribution match.
- [ ] Built-in artifact-stack layers expose no contribution provenance (`layer`, `sourceId`, and `lookupId` are null) and round-trip through the public `{kind}:{name}` artifact ID.
- [ ] Identical manifest coordinates produce identical IDs across processes, machines, project locations, and reinstalls.
- [ ] Manifest-backed IDs do not depend on artifact contents, timestamps, manifest hashes, archive paths, or installation directories.
- [ ] Existing `name` fields and name-based resolution remain unchanged.
- [ ] Tests cover each provenance-backed layer and artifact kind, hook uniqueness, resolver repeatability, manifest lookup round-trips, built-in public-ID round-trips, and backward compatibility.

### Additional Context

This request defines the ID contract only. Adding the JSON output surfaces themselves for `preset info --json`, `extension info --json`, or `specify artifact` is out of scope and will be handled separately. Changing precedence, resolution, installation, or uninstall behavior is also out of scope. Relevant implementation areas include the extension manifest schema and validation and `PresetResolver` composition handling. IDs must not expose install paths, secrets, or connection strings.

### PR DESCRIPTION
**Pull request by nicolehaugen**
Fixes #4210

## Summary

Every command, template, script, and hook contribution returned by preset and extension manifest surfaces now carries a computed opaque `id`, and every resolved artifact-stack layer carries a matching `lookupId` derived from the same recipe.

### Identifier contract

- Named contributions: `{layer}:{sourceId}:{kind}:{name}` — `layer` ∈ `core|preset|extension`, `sourceId` = `_` for core / preset id / extension id, `kind` ∈ `command|template|script|hook`, `name` = the declared name.
- Hook contributions: `{layer}:{sourceId}:hook:{eventName}:{command}`, with a `:{12-hex}` discriminator appended when two entries in the same source share the same `(eventName, command)` pair. Discriminator = first 12 lowercase hex chars of `sha256(canonical_json(entry − {eventName, command}))`. Two hook entries whose remaining declared fields are byte-identical are rejected at manifest load — there is no meaningful way to distinguish them.
- `:` is reserved as the component separator and is now guarded on hook `eventName`/`command`. All other id-component fields are already regex-constrained to forbid it.
- Identifiers are computed at read/serialization/resolution time from author-declared manifest content only — no paths, timestamps, `os.environ`, or file-content hashes contribute — so they are stable across machines, reinstalls, and directory moves. Nothing is persisted.
- Project-local overrides carry a resolver-only sentinel `project:_:{kind}:{name}` `lookupId` that intentionally does not match any manifest contribution.

The change is purely additive: existing `name`-based resolution is preserved, and no production call-site keys off the new fields.

### Modules touched

- `src/specify_cli/_identifier.py` — new pure derivation module (`derive_named_id`, `derive_hook_id`, `hook_discriminator`, `canonical_json`, `validate_component`, `PROJECT_OVERRIDE_LAYER`).
- `src/specify_cli/extensions/__init__.py` — hook `:` guards on `eventName` and `command`, byte-identical hook duplicate rejection, `ExtensionManifest.iter_contributions()` and `ExtensionManifest.contribution_id()`.
- `src/specify_cli/presets/__init__.py` — `PresetManifest.iter_contributions()` and `PresetManifest.contribution_id()`, `lookupId` on every layer emitted by `PresetResolver.collect_all_layers()`.

### Tests added

- `tests/test_contribution_ids.py` — 39 tests covering the layer×kind derivation matrix, canonical JSON semantics, hook discriminator (no-collision, collision-with-suffix, reorder-stability, byte-identical rejection), `:` component guards, contribution-surface shape (additive, underlying data un-mutated), resolver `lookupId` round-trip for `project`/`core`/`preset` layers, cross-subprocess determinism, mtime independence, and non-persistence.

### Docs updated

- `extensions/EXTENSION-API-REFERENCE.md` — new "Contribution Identifiers" section (grammar, hook convention, discriminator recipe, reserved character, `project:` sentinel, Python API, determinism guarantees, opacity guidance) and TOC entry.
- `docs/reference/presets.md` — new "Contribution Identifiers" section that cross-links to the extension reference for the full grammar.

### Validation

`pytest -q` on the affected suites is clean apart from 10 pre-existing symlink-related failures caused by Windows privilege limitations (they fail identically on `main` without these changes). All 39 new tests pass; all extensions/presets/hooks/unit tests unaffected by symlinks pass.

### Deviations from tasks.md

The scratch task list was drafted against an idealized layout that does not match the repo. Concrete adjustments applied in place:

- Identifier module lives at `src/specify_cli/_identifier.py` (top-level helper, matching the `_download_security.py` / `_utils.py` pattern) rather than an idealized `src/specify_cli/manifests/_identifier.py`. There is no shared `manifests/` package here.
- Manifest validation guards and hook duplicate detection live inside the existing `_validate()` methods on `ExtensionManifest` (and, where relevant, `PresetManifest`) rather than in a separate `manifests/validation.py` / `manifests/loader.py`.
- Contribution surfaces are exposed as new `iter_contributions()` / `contribution_id()` methods returning dicts rather than as typed `ContributionRef` / `HookContribution` / `ResolvedStackLayer` dataclasses. Existing manifest storage stays untyped and byte-for-byte unchanged.
- Layer dicts returned by `PresetResolver.collect_all_layers()` gain a `lookupId` key. There is no `ResolvedStackLayer` type to add a field to.
- Test fixtures are built programmatically in the test module rather than checked in as an on-disk fixture tree; this matches the existing pattern in `tests/test_extensions.py` / `tests/test_presets.py`.
- The pre-feature pytest count sentinel is skipped — it is a CI heuristic, not a correctness check, and produces false positives on any unrelated test addition. Backward compatibility is asserted directly by leaving every existing test unmodified and by explicit additive-shape tests on the enriched contribution dicts.

### CODE REVIEW

File: src/specify_cli/extensions/__init__.py [hunk 1]
```diff
@@ -437,6 +456,35 @@ def _validate(self):
                                 f"Hook '{hook_name}' has invalid 'priority': "
                                 "must be >= 1"
                             )
+                    event_entries.append(entry)
+
+                # Reject two hook entries under the same (event, command) whose
+                # declared fields (with eventName/command stripped) canonicalize
+                # to the same byte string — those are semantically identical
+                # listeners with no way to address them separately.
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply src/specify_cli/extensions/__init__.py]: @copilot consider this fix: **The idea:** in `iter_contributions()`, apply the same "last-write-wins by `(eventName, command)`" collapse the installer already does, *before* computing IDs. Then IDs are only issued for hooks that will actually exist on disk. No ghost IDs, no discriminator machinery needed for hooks.


**Concretely**, replace the flatten step (around lines 828–842) with something like:


Python




```
hooks = self.hooks or {}
collapsed: List[tuple[str, dict]] = []
seen: Dict[tuple[str, str], int] = {}   # (event, command) -> index in collapsed
for event_name, hook_config in hooks.items():
    for entry in coerce_hook_entries(hook_config):
        if not isinstance(entry, dict):
            continue
        command = entry.get("command")
        if not command:
            continue
        normalized = dict(entry)
        normalized.setdefault("eventName", event_name)
        key = (event_name, command)
        if key in seen:
            collapsed[seen[key]] = (event_name, normalized)   # last wins
        else:
            seen[key] = len(collapsed)
            collapsed.append((event_name, normalized))

```





Then iterate `collapsed` to build IDs. Since each `(event, command)` is now unique, `derive_hook_id` no longer needs the sibling/discriminator arguments — you can pass empty sibling list and empty declared_fields (or simplify the call), and the ID becomes purely a function of `(source, event, command)`, which is exactly what the installer keys on. Round-trips perfectly.

File: src/specify_cli/extensions/__init__.py [hunk 2]
```diff
@@ -725,6 +744,110 @@ def hooks(self) -> Dict[str, Any]:
...
+                normalized = dict(entry)
+                normalized.setdefault("eventName", event_name)
+                key = (event_name, command_value)
+                if key in seen:
+                    collapsed[seen[key]] = (event_name, normalized)
+                else:
+                    seen[key] = len(collapsed)
+                    collapsed.append((event_name, normalized))
+
+        for event_name, entry in collapsed:
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply src/specify_cli/extensions/__init__.py]: @copilot This should behave equivalent to the behavior of the installer - also, is there existing functions/logic that you can reuse?

File: docs/reference/presets.md [hunk 3]
```diff
@@ -205,6 +205,25 @@ specify preset add team-workflow --priority 10
...
+- `layer` is one of `core`, `preset`, or `extension`.
+- `sourceId` is `_` for `core`, the preset pack id for `preset`, or the extension id for `extension`.
+- `kind` is one of `command`, `template`, or `script`.
+- `name` is the entry's declared `name` field.
+
+Identifiers are computed on demand from author-declared manifest content and are never persisted to `.specify/` or any cache. Copying a preset to another machine (or touching its files) does not change the identifiers it produces.
+
+`PresetResolver.collect_all_layers()` returns layer dicts that each include a `lookupId` field pointing back to the originating contribution's `id`. Project-local overrides in `.specify/templates/overrides/` are a resolver-only concept — they carry a synthetic `project:_:{kind}:{name}` `lookupId` that intentionally does not match any manifest contribution.
+
+For the full grammar, including the hook name-component convention and the discriminator recipe used by extensions, see the [Extension API Reference — Contribution Identifiers](../../extensions/EXTENSION-API-REFERENCE.md#contribution-identifiers) section.
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply docs/reference/presets.md]: @copilot fix this issue

**Comment by nicolehaugen** [comment]
@copilot **Apply `validate_component` to all named contribution `name` fields at manifest-load time.**

Today it's only called for hook event names and hook commands. That leaves a hole where a preset or extension can declare a contribution `name` containing `:`, which produces a malformed identifier that consumers cannot unambiguously split.

Specifically, add a `validate_component(entry["name"], "<field label>")` call in:

- **`src/specify_cli/presets/__init__.py`** — where `provides.templates[]` entries (kind `command`/`template`/`script`) are validated.
- **`src/specify_cli/extensions/__init__.py`** — where `provides.templates[]` and `provides.scripts[]` names are validated. Extension command names already go through `EXTENSION_COMMAND_NAME_PATTERN` which excludes `:`, so those are covered, but adding the explicit call keeps the guarantee uniform.

Wrap each in `try/except IdentifierComponentError` and re-raise as `ValidationError` to match the existing pattern used for hook fields.

File: src/specify_cli/extensions/__init__.py [hunk 4]
```diff
@@ -213,6 +219,27 @@ def coerce_hook_entries(hook_config: Any) -> List[Any]:
     return hook_config if isinstance(hook_config, list) else [hook_config]
 
 
+def collapse_hook_event_entries(event_name: str, hook_config: Any) -> List[Dict[str, Any]]:
+    """Collapse a hook event to final entries using installer last-write-wins.
+
+    Duplicate commands are removed and re-inserted so the final declaration is
+    retained at the end of the event list, matching register-time ordering.
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply src/specify_cli/extensions/__init__.py:226]: @copilot address this issue

File: docs/reference/presets.md [hunk 5]
```diff
@@ -205,6 +205,25 @@ specify preset add team-workflow --priority 10
...
 
+## Contribution Identifiers
+
+Every command, template, and script contributed by a preset (or an extension, or the core layer) is addressable at read time by a deterministic opaque identifier of the form:
+
+```text
+{layer}:{sourceId}:{kind}:{name}
+```
+
+- `layer` is one of `core`, `preset`, or `extension`.
```
Comment (nicolehaugen) [inline docs/reference/presets.md:216]: @copilot clarify this: 

| Layer | `sourceId` value | Where it comes from |
| --- | --- | --- |
| `core` | `"_"` (literal underscore) | Placeholder — core has no manifest, so there's no id to use |
| `preset` | The preset pack's `id` | Top-level `id:` field in `preset.yml` (e.g., `spec-kit-defaults`) — same value used throughout `PresetManifest.id` and the `presets/<pack_id>/` install directory name |
| `extension` | The extension's `id` | `extension.id` field in `extension.yml` (e.g., `speckit.git`) — same value used throughout `ExtensionManifest.id` and the `extensions/<ext_id>/` install directory name=> |

The point is: **`sourceId` is whatever the manifest declares as its own id**, and it's the same string used everywhere else in the system to refer to that preset/extension (directory names, install manifests, `hooks.json` entries, etc.). That's what makes the identifier a stable join key back to the originating source.

File: docs/reference/presets.md [hunk 6]
```diff
@@ -205,6 +205,25 @@ specify preset add team-workflow --priority 10
...
+- `sourceId` is `_` for `core`, the preset pack id for `preset`, or the extension id for `extension`.
+- `kind` is one of `command`, `template`, or `script`.
```
Comment (nicolehaugen) [inline docs/reference/presets.md]: Are hooks treated separately?

File: extensions/EXTENSION-API-REFERENCE.md [hunk 7]
```diff
@@ -859,7 +860,52 @@ satisfied = version_satisfies("1.2.3", ">=1.0.0,<2.0.0")  # bool
...
+
+### The `project:` sentinel
+
+Project-local overrides in `.specify/templates/overrides/` are a resolver-only concept — they have no backing manifest and cannot appear in `iter_contributions()`. Layers of that kind carry a synthetic `lookupId` of the form `project:_:{kind}:{name}` so consumers that reverse-lookup the id always see "not found", which is the intended behaviour: overrides are addressable at the stack level, not as first-class contributions.
+
+### Python API
+
+`ExtensionManifest.iter_contributions()` yields dicts of the form `{layer, sourceId, kind, name, id, ...author-declared fields}`; each entry's `id` is the computed identifier. `ExtensionManifest.contribution_id(kind, name)` returns the id for a single lookup, or `None` if no contribution matches. `PresetManifest` exposes the same two methods.
```
Comment (nicolehaugen) [inline extensions/EXTENSION-API-REFERENCE.md]: Style nit for consistency with the rest of this file: the surrounding sections (`Python API`, `Version Functions`) document APIs with a fenced signature block, not prose. Consider matching that convention here:

```python
class ExtensionManifest:
    def iter_contributions(self) -> list[dict]: ...
    def contribution_id(self, kind: str, name: str) -> str | None: ...

class PresetManifest:
    def iter_contributions(self) -> list[dict]: ...
    def contribution_id(self, kind: str, name: str) -> str | None: ...
```

followed by a short note that each dict carries `{layer, sourceId, kind, name, id, ...author-declared fields}` and that `contribution_id` returns `None` on no match. Content is accurate as-is, purely a formatting call.

File: tests/test_contribution_ids.py [hunk 8]
```diff
@@ -0,0 +1,514 @@
+"""Tests for the deterministic contribution-id and stack lookup-id feature.
```
Comment (nicolehaugen) [inline tests/test_contribution_ids.py:1]: drop tests test\_named\_id\_stable\_across\_two\_derivations, `test_representation_shape_is_additive_for_preset` / `_for_extension`, **`test_identifiers_match_across_subprocesses`**, test\_ids\_independent\_of\_paths\_and\_mtimes

**Review by nicolehaugen — COMMENTED** [review]
@copilot address this feedback

File: docs/reference/presets.md (hunk 6 above)
Comment (nicolehaugen) [reply docs/reference/presets.md]: @copilot change line 218 to `kind is one of command, template, or script (see below for hooks)` so the reader isn't surprised.

File: tests/test_contribution_ids.py [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+
+Every command / template / script / hook contribution surfaced by a preset or
+extension manifest exposes a computed ``id`` derived from author-declared data
+only, and every layer of a resolved artifact stack exposes a matching
+``lookupId``. The scenarios below cover: the identifier grammar across every
+``layer x kind`` combination, duplicate hook collapse behavior, cross-process
+byte-stability, path/mtime independence, and the additive-only shape guarantee
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply tests/test_contribution_ids.py]: @copilot remove the claim that is unsupported

**Comment by nicolehaugen** [comment]
@copilot for presets.md change line 218 to `kind is one of command, template, or script (see below for hooks)` so the reader isn't surprised.

File: src/specify_cli/presets/__init__.py [hunk 10]
```diff
@@ -5611,6 +5664,9 @@ def _find_in_subdirs(base_dir: Path) -> Optional[Path]:
                     "strategy": "replace",
                     "extension_id": ext_id,
                     "extension_dir": ext_dir,
+                    "lookupId": derive_named_id(
+                        "extension", ext_id, template_type, template_name
+                    ),
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply src/specify_cli/presets/__init__.py:5673]: @copilot `lookupId` for extension layers is using the extension directory/registry key (`ext_id`) as `sourceId`, but manifest-backed contribution IDs use `ExtensionManifest.id` (`extension.id` from `extension.yml`). These can differ for unregistered/local-copy extension dirs, causing join failure (`lookupId` != `contribution_id()`).


Please fix by using the loaded manifest id for `sourceId` whenever `_extension_manifest_declared_template()` returns a declared manifest entry, and only using directory key for convention/fallback layers with no manifest declaration. `lookupId` for extension layers is using the extension directory/registry key (`ext_id`) as `sourceId`, but manifest-backed contribution IDs use `ExtensionManifest.id` (`extension.id` from `extension.yml`). These can differ for unregistered/local-copy extension dirs, causing join failure (`lookupId` != `contribution_id()`).


Please fix by using the loaded manifest id for `sourceId` whenever `_extension_manifest_declared_template()` returns a declared manifest entry, and only using directory key for convention/fallback layers with no manifest declaration. Also add a regression test where folder name and `extension.id` differ to prove round-trip still works.

File: extensions/EXTENSION-API-REFERENCE.md [hunk 11]
```diff
@@ -859,7 +860,65 @@ satisfied = version_satisfies("1.2.3", ">=1.0.0,<2.0.0")  # bool
...
+```text
+{layer}:{sourceId}:{kind}:{name}
+```
+
+- `layer` is one of `core`, `preset`, or `extension`.
+- `sourceId` is `_` for `core`, the preset pack id for `preset`, or the extension id for `extension`.
+- `kind` is one of `command`, `template`, `script`, or `hook`.
+- `name` is the contribution's declared `name` field.
+
+`sourceId` is the source's own stable system identifier: `_` for the manifest-less core layer, `preset.id` from `preset.yml` for presets, and `extension.id` from `extension.yml` for extensions. It is the same string used elsewhere to refer to that preset or extension (install directories, registries, resolver metadata, and hook metadata), which makes identifiers stable join keys back to their originating source.
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply extensions/EXTENSION-API-REFERENCE.md]: @copilot fix

File: docs/reference/presets.md [hunk 12]
```diff
@@ -205,6 +205,35 @@ specify preset add team-workflow --priority 10
...
+- `kind` is one of `command`, `template`, or `script` (see below for hooks).
+- `name` is the entry's declared `name` field.
+
+`sourceId` is the source's own stable system identifier:
+
+| Layer | `sourceId` value | Where it comes from |
+| --- | --- | --- |
+| `core` | `_` (literal underscore) | Placeholder because core has no manifest id |
+| `preset` | The preset pack's `id` | The manifest's `preset.id` field — the same value used by `PresetManifest.id`, the install directory, registries, and resolver layer metadata |
+| `extension` | The extension's `id` | The manifest's `extension.id` field — the same value used by `ExtensionManifest.id`, the install directory, registries, and hook metadata |
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply docs/reference/presets.md]: @copilot fix

File: src/specify_cli/extensions/__init__.py [hunk 13]
```diff
@@ -725,6 +783,91 @@ def hooks(self) -> Dict[str, Any]:
...
+        the identifiers reflecting the current in-memory manifest state.
+        """
+        source_id = self.id
+        contributions: List[Dict[str, Any]] = []
+
+        for cmd in self.commands:
+            enriched = dict(cmd)
+            name = cmd.get("name", "")
+            enriched.update(
+                layer="extension",
```
[the remark this answers is not in the document]
Comment (nicolehaugen) [reply src/specify_cli/extensions/__init__.py:876]: @copilot please align `ExtensionManifest` validation with existing install-time command/alias semantics instead of adding separate duplicate-ID handling.


Right now duplicate command/alias validation lives in `ExtensionManager._collect_manifest_command_names()`, so `ExtensionManifest(...)` can be constructed and `iter_contributions()` can expose duplicate command IDs before install-time validation rejects the manifest. Extract the existing command/alias name collection + duplicate validation into a shared helper, then call it from both:



1. `ExtensionManifest._validate()` after command names and aliases have been normalized/corrected.

2. `ExtensionManager._collect_manifest_command_names()` / install conflict validation.



This should reject duplicate primary command names, primary-command-vs-alias collisions, and alias-vs-alias collisions consistently at manifest-load time, matching what install already does. Do **not** collapse duplicate commands; commands should remain invalid when duplicated. Please include targeted regression tests for duplicate primary command names and at least one alias collision that previously only failed during install validation.

**Comment by nicolehaugen** [comment]
Superseded by #4305, which consolidates this work with #4267 into a single PR against `main`. The identifier grammar here and the resolver changes in #4267 evolved together through review — splitting them cleanly would leave both PRs with incomplete stories, so ship as one.
